File: server/app/models/bill.rb
class Bill < ApplicationRecord
  has_many :participations
  accepts_nested_attributes_for :participations

  belongs_to :added_by, class_name: 'Member'
  belongs_to :group

  validates_presence_of :title, :added_by, :group

  def serialize
    {
      title: title,
      added_by: added_by,
      participants: participations.map do |p|
        {
          name: p.member.name,
          amount: p.amount
        }
      end
    }
  end
end

File: server/app/models/bill.js
import mongoose from 'mongoose'
const Schema = mongoose.Schema
const ObjectId = mongoose.Schema.Types.ObjectId

const BillSchema = new Schema({
  date: Date,
  title: String,
  paid: {
    amount: Number,
    userId: ObjectId,
  },
  addedBy: ObjectId,
  groupId: ObjectId,
  participants: [ObjectId],
})

BillSchema.method({})
BillSchema.static({})
mongoose.model('Bill', BillSchema)
